src/plugins/discover/public/application/main/components/top_nav/on_save_search.tsx

```diff
@@ -106,31 +106,38 @@ export async function onSaveSearch({
   onClose?: () => void;
   onSaveCb?: () => void;
 }) {
-  const { uiSettings } = services;
+  const { uiSettings, savedObjectsTagging } = services;
   const onSave = async ({
     newTitle,
     newCopyOnSave,
     newTimeRestore,
     newDescription,
+    newTags,
     isTitleDuplicateConfirmed,
     onTitleDuplicate,
   }: {
     newTitle: string;
     newTimeRestore: boolean;
     newCopyOnSave: boolean;
     newDescription: string;
+    newTags: string[];
     isTitleDuplicateConfirmed: boolean;
     onTitleDuplicate: () => void;
   }) => {
     const currentTitle = savedSearch.title;
     const currentTimeRestore = savedSearch.timeRestore;
     const currentRowsPerPage = savedSearch.rowsPerPage;
+    const currentDescription = savedSearch.description;
```

Comment:
this for fixing this! just load thinking that we could improve this code  unrelated to this PR, and no priority ... but extracting values of the given savedSearch, mutating the object, continue to mutate it further when preparing to save, and if it fails, we unset values ... we should better provide a saved search for saving, and if it's successful we continue operations to modify / change our saved search, rather than undoing changes

Reply:
I fully agree, that would be much less error prone I think.

Reply:
Created an issue for this: #139511.